### ISSUE TITLE
Load dataset in memory when training

### ISSUE THREAD

**Issue by Zhw098** [OpenNMT/OpenNMT-py#2126]
I tried to train an retrosynthesis model using OpenNMT
But during the training process,the dataset will be loaded several times. And GPU utilization is very low
Installing a higher version still does not solve the  problem.    @francoishernandez
Thanks!
pytorch 1.5 python 3.6 OpenNMT 0.4.1
`training command: CUDA_VISIBLE_DEVICES=1
python  train.py -data data/${dataset}/${dataset} \
                   -save_model experiments/checkpoints/${dataset}/${dataset}_model \
                   -seed 42 -gpu_ranks 0  -batch_size 4096   -accum_count 4 \`
  
![image](https://user-images.githubusercontent.com/81342396/144017244-fc777114-fdc5-4e46-b81d-5f18174028fe.png)
              

![image](https://user-images.githubusercontent.com/81342396/144015918-2ff2073b-23bf-4650-b29b-cfc0685293b7.png)

**Comment by francoishernandez** [comment]
Duplicate of #1930, where I give some pointers if you want to contribute.

**Comment by Zhw098** [comment]
> Duplicate of #1930, where I give some pointers if you want to contribute.

Thanks for your reply. but I'm a new user. Can you be more detailed?
Hope your helps.
Thanks！
Willow
